aiohttp/web_reqrep.py

```diff
@@ -240,6 +240,22 @@ def if_modified_since(self, _IF_MODIFIED_SINCE=hdrs.IF_MODIFIED_SINCE):
         return None
 
     @reify
+    def authorization(self, _AUTHORIZATION=hdrs.AUTHORIZATION):
+        """Parse the Authorization header and return the username and password,
+        or ``None`` if the header is absent or cannot be parsed.
+
+        :rtype:  :class:`aiohttp.helpers.BasicAuth`
+
+        """
+        header = self.headers.get(_AUTHORIZATION)
+        if header is not None:
+            try:
+                return BasicAuth.decode(header)
```

Comment:
@luhn I would agree with that with happy, but you don't know how many services that works with the local national encoding by default (this includes latin1, cp1251, etc.) /: 

The charset parameter you reference works only with the flow:
1. Make request
2. Receive HTTP 401 with WWW-Authenticate and that charset param
3. Resend request with Authorization header where credentials respects previously mentioned charset

This means some interaction with the user and works perfectly with the browsers, but not with the libraries ):